From 358cd8888d8cfa0e2565cfe8e5a001ae9e43915f Mon Sep 17 00:00:00 2001
From: andy3_kuo <andy3_kuo@asus.com>
Date: Thu, 12 Nov 2020 19:00:35 +0800
Subject: [PATCH] QCA6174: disable P2P & WFD

---
 Kbuild | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/Kbuild b/Kbuild
index a3be8a3b5..895593249 100644
--- a/Kbuild
+++ b/Kbuild
@@ -1072,7 +1072,6 @@ CDEFINES :=	-DANI_LITTLE_BYTE_ENDIAN \
 		-DWLAN_FEATURE_PACKET_FILTERING \
 		-DWLAN_FEATURE_VOWIFI \
 		-DWLAN_FEATURE_11AC \
-		-DWLAN_FEATURE_P2P_DEBUG \
 		-DWLAN_ENABLE_AGEIE_ON_SCAN_RESULTS \
 		-DWLANTL_DEBUG\
 		-DWLAN_NS_OFFLOAD \
@@ -1278,8 +1277,6 @@ CDEFINES += -DNBUF_MAP_UNMAP_DEBUG
 endif
 
 ifeq ($(HAVE_CFG80211),1)
-CDEFINES += -DWLAN_FEATURE_P2P
-CDEFINES += -DWLAN_FEATURE_WFD
 ifeq ($(CONFIG_QCOM_VOWIFI_11R),y)
 CDEFINES += -DKERNEL_SUPPORT_11R_CFG80211
 CDEFINES += -DUSE_80211_WMMTSPEC_FOR_RIC
@@ -1614,7 +1611,7 @@ else #CONFIG_MOBILE_ROUTER
 
 #Open P2P device interface only for non-Mobile router use cases
 ifneq ($(CONFIG_SUPPORT_IFTYPE_P2P_DEVICE_VIF), y)
-CDEFINES += -DWLAN_OPEN_P2P_INTERFACE
+#CDEFINES += -DWLAN_OPEN_P2P_INTERFACE
 endif
 
 #Enable 2.4 GHz social channels in 5 GHz only mode for p2p usage
-- 
2.17.1

