From f5c3dc980f6902a5e86b0feb4f998f67912d7b9e Mon Sep 17 00:00:00 2001
From: andy3_kuo <andy3_kuo@asus.com>
Date: Fri, 7 May 2021 00:23:23 +0800
Subject: [PATCH 5/5] CORE: bypass load athsetup.bin for load firmare fail

---
 CORE/SERVICES/BMI/ol_fw.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/CORE/SERVICES/BMI/ol_fw.c b/CORE/SERVICES/BMI/ol_fw.c
index be4588b94..3c8d7261b 100644
--- a/CORE/SERVICES/BMI/ol_fw.c
+++ b/CORE/SERVICES/BMI/ol_fw.c
@@ -1166,8 +1166,9 @@ defined(CONFIG_NON_QC_PLATFORM_PCI)
 #ifdef QCA_SIGNED_SPLIT_BINARY_SUPPORT
 			bin_sign = TRUE;
 #endif
-			printk(KERN_INFO "%s: Loading setup file %s\n",
+			printk(KERN_INFO "%s: bypass Loading setup file %s\n",
 					__func__, filename);
+			return -1;
 #endif /* CONFIG_CNSS */
 		} else {
 			printk("%s: no Setup file needed\n", __func__);
-- 
2.25.1

