From a55d18a569fdf6171e22fcb6aa40aaef1aab08b1 Mon Sep 17 00:00:00 2001
From: andy3_kuo <andy3_kuo@asus.com>
Date: Tue, 6 Jul 2021 23:08:41 +0800
Subject: [PATCH] imx-mkimage: imx8mq : add 2G support

---
 iMX8M/soc.mak | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/iMX8M/soc.mak b/iMX8M/soc.mak
index bc6acc1..89a3909 100644
--- a/iMX8M/soc.mak
+++ b/iMX8M/soc.mak
@@ -44,7 +44,7 @@ else
 PLAT = imx8mq
 HDMI = yes
 SPL_LOAD_ADDR = 0x7E1000
-TEE_LOAD_ADDR = 0xfe000000
+TEE_LOAD_ADDR = 0xbe000000
 ATF_LOAD_ADDR = 0x00910000
 VAL_BOARD = arm2
 #define the F(Q)SPI header file
-- 
2.25.1

